src/semantic_csv/core.clj

```diff
@@ -149,7 +149,75 @@
 ;; <br/>
 ;; Next, let's observe that `:this` and `:that` point to strings, while they should really be pointing to
 ;; numeric values.
-;; This can be addressed with the following function:
+;; This can be addressed with the following functions:
+
+
+;; ## cast-rules
+
+;; cast rules to be used in sniff-data
+;; this map should have a structure as such
+;; {:cast-class {:hierarch [:cast-type-1 :cast-type-2 ...]
+;;               :cast-type-1
+;;               :cast-type-2
+;;               ...}}
+;; The cast class defines the class of values (i.e. numeric values or date/time values).
+;; The cast type defines the semantic type within that class (i.e. integer, decimal, rational)
+(def cast-rules
```

Comment:
@mahinshaw I'd suggest `default-cast-rules` for this.

Reply:
Agree